gtsam/slam/EssentialMatrixWithCalibrationFactor.h

```diff
@@ -0,0 +1,119 @@
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file EssentialMatrixWithCalibrationFactor.h
+ *
+ * @brief A factor evaluating algebraic epipolar error with essential matrix and calibration as variables.
+ *
+ * @author Ayush Baid
+ * @author Akshay Krishnan
+ * @date April 23, 2021
+ */
+
+#pragma once
+
+#include <gtsam/nonlinear/NonlinearFactor.h>
+#include <gtsam/geometry/EssentialMatrix.h>
+#include <iostream>
+
+namespace gtsam {
+
+/**
+ * Factor that evaluates algebraic epipolar error (K^-1 p)'E (K^-1 p) for given essential matrix and calibration shared
+ * between two images.
+ */
+template<class CALIBRATION>
+class EssentialMatrixWithCalibrationFactor: public NoiseModelFactor2<EssentialMatrix, CALIBRATION > {
+
+  Point2 pA_, pB_; ///< points in pixel coordinates
+
+  typedef NoiseModelFactor2<EssentialMatrix, CALIBRATION> Base;
+  typedef EssentialMatrixWithCalibrationFactor This;
+
+public:
+
+  /**
+   *  Constructor
+   *  @param essentialMatrixKey Essential Matrix variable key
+   *  @param calibrationKey Calibration variable key
+   *  @param pA point in first camera, in pixel coordinates
+   *  @param pB point in second camera, in pixel coordinates
+   *  @param model noise model is about dot product in ideal, homogeneous coordinates
+   */
+  EssentialMatrixWithCalibrationFactor(Key essentialMatrixKey, Key calibrationKey, const Point2& pA, const Point2& pB,
+      const SharedNoiseModel& model) :
+      Base(model, essentialMatrixKey, calibrationKey), pA_(pA), pB_(pB) {}
+
+
+  /// @return a deep copy of this factor
+  gtsam::NonlinearFactor::shared_ptr clone() const override {
+    return boost::static_pointer_cast<gtsam::NonlinearFactor>(
+        gtsam::NonlinearFactor::shared_ptr(new This(*this)));
+  }
+
+  /// print
+  void print(const std::string& s = "",
+      const KeyFormatter& keyFormatter = DefaultKeyFormatter) const override {
+    Base::print(s);
+    std::cout << "  EssentialMatrixWithCalibrationFactor with measurements\n  ("
+        << pA_.transpose() << ")' and (" << pB_.transpose() << ")'"
+        << std::endl;
+  }
+
+  /// vector of errors returns 1D vector
+  /**
+   * @brief Calculate the algebraic epipolar error p' (K^-1)' E K p.
+   *
+   * @param E essential matrix for key essentialMatrixKey
+   * @param K calibration (common for both images) for key calibrationKey
+   * @param H1 optional jacobian in E
+   * @param H2 optional jacobian in K
+   * @return * Vector
+   */
+  Vector evaluateError(const EssentialMatrix& E, const CALIBRATION& K,
+    boost::optional<Matrix&> H1 = boost::none, boost::optional<Matrix&> H2 = boost::none) const override {
+    Vector error(1);
+    // converting from pixel coordinates to normalized coordinates cA and cB
+    Matrix cA_H_K; // dcA/dK
```

Comment:
Use fixed size

Reply:
It will depend on the number of params for K, Hence I wont be able to use a fixed size.

Reply:
Not true at all: CALIBRATION is known at compile time

Reply:
Changed it. Thanks!